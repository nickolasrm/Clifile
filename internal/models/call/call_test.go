package call_test

import (
	"testing"

	"github.com/nickolasrm/clifile/internal/models/call"
	. "github.com/nickolasrm/clifile/test/util"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

func TestCall(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Call Suite")
}

var _ = Describe("Call", func() {
	When("a newcall is created", func() {
		var c *call.Call

		BeforeEach(func() {
			c = call.NewCall("name", "func", map[string]string{
				"arg1": "val",
				"arg2": "val2",
			})
		})

		It("should return a call type", func() {
			Expect(c).ToNot(BeNil())
		})
		When("Function() is called", func() {
			It("should return the function name", func() {
				MatchSnapshot(c.Name())
			})
		})
		When("Name() is called", func() {
			It("should return the call name", func() {
				MatchSnapshot(c.Function())
			})
		})
		When("Arguments() is called", func() {
			It("should return the arguments map", func() {
				MatchSnapshot(c.Arguments())
			})
		})
		When("Argument() is called", func() {
			When("an existing argument is passed", func() {
				It("should return its value", func() {
					MatchSnapshot(c.Argument("arg1"))
				})
			})
			When("an non-existing argument is passed", func() {
				It("should be empty", func() {
					Expect(c.Argument("random")).To(BeEmpty())
				})
			})
		})
	})
})
