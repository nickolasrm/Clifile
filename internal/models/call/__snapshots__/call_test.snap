
[Call when a newcall is created when Function() is called should return the function name - 1]
[]interface {}{
    "name",
}
---

[Call when a newcall is created when Name() is called should return the call name - 1]
[]interface {}{
    "func",
}
---

[Call when a newcall is created when Arguments() is called should return the arguments map - 1]
[]interface {}{
    map[string]string{"arg1":"val", "arg2":"val2"},
}
---

[Call when a newcall is created when Argument() is called when an existing argument is passed should return its value - 1]
[]interface {}{
    "val",
}
---
